dietpi/dietpi-software

```diff
@@ -3795,6 +3795,12 @@ _EOF_
 
 			Banner_Installing
 			AGI redis-server
+			#Redis php module | == 2 to check for existing installs, if == 1, then module will be installed together with PHP, to prevent dependency installations
+			if (( ${aSOFTWARE_INSTALL_STATE[89]} == 2 )); then
```

Comment:
This will only be true, if ```dietpi-software``` has previously installed ```89``` and rebooted.
```=2``` is not set until the very end of dietpi-software installs:
https://github.com/Fourdee/DietPi/blob/testing/dietpi/dietpi-software#L14286-L14296

```=1``` during installations, ```=2``` set at the very end.

Reply:
Yes this is okay: if redis and php were chosen to install together, php part includes installation of php(5)-redis anyway.

This was just necessary, if php was already installed before and you somehow chose to install redis-server later, e.g. you read about advantages with ownCloud/Nextcloud 😉.

The problem with using ">=1" is, as I tried to explain in code,  that redis-server package installation is done before PHP, and as php-redis depends on php packages, it will install them and in case choose some random optional dependency (libapache-mod-php / php-fpm / php-cgi e.g.): https://packages.debian.org/stretch/php-redis

We could switch positions of redis and php installation as another solution. Generally there would be the same problem then with php-redis' server dependency, but redis-server is so far just suggestion, not dependency: https://packages.debian.org/jessie/php5-redis

Long story short: On choosing redis (==1), it will install the module, if php==2 and if php==1 then php loop will see redis==1 and take care it's module installation.

Reply:
@MichaIng 

Thanks , understood 👍 

I'll get this merged.